x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealm.java

```diff
@@ -126,9 +126,15 @@ public JwtRealm(final RealmConfig realmConfig, final SSLService sslService, fina
             this.httpClient = null; // no setting means no HTTP client
         }
 
-        this.jwksAlgsHmac = this.parseJwksAlgsHmac();
-        this.jwksAlgsPkc = this.parseJwksAlgsPkc();
-        this.verifyAnyAvailableJwkAndAlgPair();
+        // If HTTPS client was created in JWT realm, any exception after that point requires closing it to avoid a thread pool leak
+        try {
+            this.jwksAlgsHmac = this.parseJwksAlgsHmac();
+            this.jwksAlgsPkc = this.parseJwksAlgsPkc();
+            this.verifyAnyAvailableJwkAndAlgPair();
+        } catch (Throwable t) {
+            this.close();
+            throw t;
+        }
```

Comment:
Answer my own question here. I think this is added to production code because the instantiation may fail in tests and there will no instance of `jwtRealm` to call `close` upon. Generally I don't like change production code for tests. But this seems to be minor enough and the behaviour is retrow anyway.